pyomo/contrib/mindtpy/util.py

```diff
@@ -499,6 +502,16 @@ def set_solver_options(opt, solve_data, config, solver_type, regularization=Fals
             opt._solver_model.parameters.mip.strategy.presolvenode.set(3)
             if config.add_regularization in {'hess_lag', 'hess_only_lag'}:
                 opt._solver_model.parameters.optimalitytarget.set(3)
+    elif solver_name == 'appsi_cplex':
+        opt.options['timelimit'] = remaining
+        opt.options['mip_tolerances_mipgap'] = config.mip_solver_mipgap
```

Comment:
Here you should be able to do something like:

```
elif isinstance(opt, appsi.base.Solver):
    opt.config.time_limit = remaining
    opt.config.mip_gap = config.mip_solver_mipgap
```

which is not solver specific. It may not matter since it looks like you need other solver specific options.